usecases/registration.go

```
@@ -136,6 +89,61 @@ func RegisterServices(sys *components.System) {
 	}
 }
 
+func confirmLeadingRegistrar(leadingRegistrar *components.CoreSystem) *components.CoreSystem {
+	resp, err := http.Get(leadingRegistrar.Url + "/status")
+	if err != nil {
+		log.Println("lost leading registrar status:", err)
+		leadingRegistrar = nil
+		return leadingRegistrar
+	}
+	// Read from resp.Body and then close it directly after
+	bodyBytes, err := io.ReadAll(resp.Body)
+	errClose := resp.Body.Close() // Close the body directly after reading from it
+	if err != nil {
+		log.Println("\rError reading response from leading registrar:", err)
+		leadingRegistrar = nil
+		return leadingRegistrar
+	}
+	if errClose != nil {
+		log.Println("Error closing the leading registrar response body:", errClose)
+	}
+	if !strings.HasPrefix(string(bodyBytes), "lead Service Registrar since") {
+		leadingRegistrar = nil
+		log.Println("lost previous leading registrar")
+	}
+	return leadingRegistrar
+}
+
+func findLeadingRegistrar(sys *components.System, leadingRegistrar *components.CoreSystem) *components.CoreSystem {
+	for _, core := range sys.CoreS {
+		if core.Name != "serviceregistrar" {
+			continue
+		}
+
+		resp, err := http.Get(core.Url + "/status")
+		if err != nil {
+			fmt.Println("error checking service registrar status:", err)
+			continue // Skip to the next iteration of the loop
+		}
+
+		// Read from resp.Body and then close it directly after
+		bodyBytes, err := io.ReadAll(resp.Body)
+		errClose := resp.Body.Close() // Close the body directly after reading from it
```

defer again and ignore error (it doesn't help us here).